packages/app/src/components/StakeForm.tsx

```diff
@@ -11,44 +12,87 @@ import { toFixedWithPrecision, toSol } from "../lib/util";
 interface AmountInputProps {
   className?: string;
   balance: BN | undefined;
+  amount: string;
   setAmount: Function;
 }
 
 const AmountInput: React.FC<AmountInputProps> = ({
   balance,
   className,
+  amount,
   setAmount,
-}) => (
-  <div className={className}>
-    <div className="p-8 pt-3 bg-background">
-      <div className="text-right">
-        Balance:{" "}
-        <span className="text-blue">
-          {balance ? toFixedWithPrecision(toSol(balance)) : "-"} SOL
-        </span>
+}) => {
+  const [isMax, setIsMax] = useState(false);
+
+  return (
+    <div className={className}>
+      <div className="flex flex-row justify-between p-8 my-auto bg-background">
+        <img src="solana-logo.png" className="h-24 my-auto" />
+        <div className="my-auto">
+          <div className="text-right">
+            Balance:{" "}
+            <span className="text-blue">
+              {balance ? toFixedWithPrecision(toSol(balance)) : "-"} SOL
+            </span>
+          </div>
+          <span className="flex flex-row mt-2">
+            <button
+              className={clx(
+                "text-xl my-auto mx-4 px-4 py-1 border-solid border-[1px] border-green rounded-md bg-green",
+                { "bg-opacity-25": !isMax }
+              )}
+              onClick={() => {
+                setIsMax((prevState) => {
+                  if (prevState) {
+                    // setAmount here creates a warning but I don't understand why, it says this is a bad setState call
```

Comment:
When we get rid of that state all problems solved :)

Reply:
Done.